app/models/person.rb

```diff
@@ -241,15 +238,52 @@ def exported_key= new_key
     serialized_public_key = new_key
   end
 
-  #database calls
+  def username
+    @username ||= owner ? owner.username : diaspora_handle.split("@")[0]
+  end
+
+  def webfinger_hash
+    {
+      acct_uri:    "acct:#{diaspora_handle}",
+      alias_url:   "#{url}people/#{guid}",
+      hcard_url:   url[0...-1] + DiasporaFederation::Engine.routes.url_helpers.hcard_path(guid),
+      seed_url:    url,
+      profile_url: profile_url,
+      atom_url:    atom_url,
+      salmon_url:  receive_url,
```

Comment:
How about cleaning up all the URL generation a bit and making it more robust:

```ruby
def uri
  @uri ||= URI.parse(url)
  @uri.dup
end

def url_to(path)
  uri.tap {uri|  uri.path = path }.to_s
end
```

And then while at it also cleaning up `clean_url` (still used, can usages be replaced?) and `update_url` below.